src/scripts/quick_reblog.js

```diff
@@ -239,6 +269,29 @@ export const main = async function () {
     ...userBlogs.map(({ name, uuid }) => Object.assign(document.createElement('option'), { value: uuid, textContent: name }))
   );
 
+  if (rememberLastBlog) {
+    const hashToUuid = {};
+    for (const { uuid } of userBlogs) {
+      const hash = await sha256(uuid);
+      uuidToHash[uuid] = hash;
+      hashToUuid[hash] = uuid;
+    }
+
+    const mainBlog = userBlogs.find(blog => blog.primary === true);
+    accountKey = uuidToHash[mainBlog.uuid];
+
+    const { [rememberedBlogStorageKey]: rememberedBlogs = {} } =
+      await browser.storage.local.get(rememberedBlogStorageKey);
+
+    const savedBlogHash = rememberedBlogs[accountKey];
+
+    if (savedBlogHash && hashToUuid[savedBlogHash]) {
+      blogSelector.value = hashToUuid[savedBlogHash];
+    }
```

Comment:
I personally find this easier to read, too; we're trying to find a matching hash, and there's not a much more direct way of doing that than `.find()` with a strict equality check.

```suggestion
    const savedBlogUuid = Object.keys(uuidToHash).find(uuid => uuidToHash[uuid] === savedBlogHash);
    if (savedBlogUuid) blogSelector.value = savedBlogUuid;
```

Reply:
Ahh, there we go. My oneliner was really awkward because I was using `entries` instead of `keys`, and... well, you can see what I did in the commented code in the initial commit; it was bad :D

Reply:
Hey, you weren't off the mark by much!

Reply:
It's interesting to analyze why the pattern here doesn't tend to occur to me! I think on some level it's how much of my process of learning JS was doing performance optimizations, and writing out uuidToHash twice doesn't intuitively feel like a reduction in "things that the code does" even though that's a totally meaningless way to characterize code.

Reply:
programmers can have little a repeat yourself, as a treat